Rails 5 & testing

clowder opened the pull request:
_Apologies in advance for such a large PR for a [small change](https://github.com/contentful/contentful_rails/pull/21/commits/745a629df3f3e3f1d4911df6d628f03ee678612b). But this was the only way I could validate the bug I was having!_

Ran into an issue with the engine trying to skip `#verify_authenticity_token` when it wasn't in the callback chain.

It seems to stem from the fact that if we inherit directly from `[ActionController::Base]` then `#protect_from_forgery` won't have been called anywhere.

dlitvakb commented:
Hey @clowder,

I'll take a look into it,

Could you please remove `Gemfile.lock` and add `*.lock` to the `.gitignore`?

Cheers

dlitvakb commented:
Also,

This won't be merged to master, as it enforces Rails 5, and most of our users are on Rails 4 or below.

But after testing, if everything goes well, I will create a `rails-5` branch with this.

Cheers

clowder commented:
Updated as requested.

The coupling to Rails 5 was pretty superficial, updated to be `> 4.2` and added travis matrix build config – first passing build [here](https://travis-ci.org/stowbotics/contentful_rails/branches).

clowder commented on contentful_rails.gemspec:
```diff
@@ -14,10 +14,9 @@ Gem::Specification.new do |s|
 
   s.add_dependency "contentful_model", ">= 0.1.7"
-  s.add_dependency "rails", ">= 4.1.8"
+  s.add_dependency "rails", ">= 4.2", "< 5.1"
```
Reason for the change here is that we aren't technically compatible with anything less than Rails 4.2 – I found it when setting up the Travis build.

We use `[ActiveSupport::Concern#class_methods]` in `[ContentfulRails::Sluggable]`, which didn't land in Rails until [4.2](http://apidock.com/rails/ActiveSupport/Concern/class_methods).

clowder commented:
Hi Team Contentful! Any guidance on getting this merged?

dlitvakb commented:
Hey @clowder,

Sorry for the delay, many tasks were prioritized over this, and I totally left it behind.

I'm looking into it now.
